packages/dynamodb-data-mapper/src/DataMapper.ts

```diff
@@ -306,6 +307,100 @@ export class DataMapper {
         }
     }
 
+    private delay(duration: number) {
+        return new Promise(resolve => setTimeout(resolve, duration));
+    }
+
+    /**
+     * Perform a UpdateTable operation using the schema accessible via the
+     * {DynamoDbSchema} property and the table name accessible via the
+     * {DynamoDbTable} property on the prototype of the constructor supplied.
+     *
+     * The promise returned by this method will not resolve until the table is
+     * active and ready for use.
+     *
+     * @param valueConstructor  The constructor used for values in the table.
+     * @param options           Options to configure the CreateTable operation
+     */
+    async createGlobalSecondaryIndex(
+        valueConstructor: ZeroArgumentsConstructor<any>,
+        {
+            readCapacityUnits,
+            streamViewType = 'NONE',
+            writeCapacityUnits,
+            indexOptions = {},
+        }: CreateTableOptions,
+        indexName: string
+    ) {
+        const schema = getSchema(valueConstructor.prototype);
+        const { attributes, indexKeys, tableKeys } = keysFromSchema(schema);
+        const TableName = this.getTableName(valueConstructor.prototype);
+
+        const globalSecondaryIndexes = indexDefinitions(indexKeys, indexOptions, schema).GlobalSecondaryIndexes;
+        const indexSearch = globalSecondaryIndexes.filter(function(index) {
+            return index.IndexName === indexName;
+        });
+        const indexDefinition: CreateGlobalSecondaryIndexAction = indexSearch[0];
+
+        var {
+            TableDescription: {TableStatus} = {TableStatus: 'UPDATING'}
+        } = await this.client.updateTable({
+            GlobalSecondaryIndexUpdates: [{
+                Create: {
+                    ...indexDefinition
+                }
+            }],
+            TableName,
+            AttributeDefinitions: attributeDefinitionList(attributes),
+        }).promise();
+
+        var remainingAttempts = 25;
+        while (TableStatus !== 'ACTIVE') {
```

Comment:
When this change is fleshed out to include more table modification commands, you'll likely want to factor this `while` loop out into a function like `function waitFor(maxAttempts: number, delayMillis: number, predicate: () => boolean): Promise<void>`